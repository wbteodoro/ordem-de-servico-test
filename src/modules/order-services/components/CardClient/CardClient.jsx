import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';


class CardClient extends Component {
  render() {

    return (
      <Card>
        <CardContent>
          {/* As informações virão da variavel dados do FormClient*/}
            <Typography color="textSecondary" gutterBottom>
              Nome: {this.props.serviceOrder.name}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              CPF: {this.props.serviceOrder.cpf}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              CNPJ: {this.props.serviceOrder.cnpf}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              Telefone: {this.props.serviceOrder.phone}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              Data de Nascimento: {this.props.serviceOrder.birth_date}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              E-mail: {this.props.serviceOrder.email}
            </Typography>
            <Typography color="textSecondary" gutterBottom>
              Nome Fantasia:{this.props.serviceOrder.fantasy_name}
            </Typography>
        </CardContent>
      </Card>      
    )
  }
}


CardClient.propTypes = {
    serviceOrder: PropTypes.object.isRequired,
}

export default CardClient