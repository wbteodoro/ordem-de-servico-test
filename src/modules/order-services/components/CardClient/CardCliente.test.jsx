import React from 'react'
import { shallow, mount } from 'enzyme'
import CardClient from './CardClient'
import { CombineLatestSubscriber } from 'rxjs/internal/observable/combineLatest';


describe('Testing CardClient Component', () => {
 
  it('should test the component rendering, if there are 7 Typographys ', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)
    expect(wrapper.find('Typography')).toHaveLength(7)
    
  })


  it('should check snapshot version', () => {    
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = shallow(<CardClient serviceOrder={serviceOrder}/>).dive()
    
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Testing the writting of the fileds', () => {
  
  it('should test if the firts field is Nome', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(0)
    // Quando acessado Typography o resultado obtido é o nome(CPF, Telefone) + o valor da  props.serviceOrder
    // Sendo assim, foi necessário separá-los e pegar o primerio elemento
    expect(findText.text().split(":")[0]).toEqual("Nome")
  })

  it('should test if the second field is CPF', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(1)
    expect(findText.text().split(":")[0]).toEqual("CPF")    
  })

  it('should test if the thrid field is CNPF', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(2)
    expect(findText.text().split(":")[0]).toEqual("CNPJ")    
  })

  it('should test if the fourth field is Telefone', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(3)
    expect(findText.text().split(":")[0]).toEqual("Telefone")    
  })

  it('should test if the fifth field is Data de Nascimento', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(4)
    expect(findText.text().split(":")[0]).toEqual("Data de Nascimento")    
  })

  it('should test if the sixth field is E-mail', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(5)
    expect(findText.text().split(":")[0]).toEqual("E-mail")    
  })

  it('should test if the seventh field is Nome Fantasia', () => {
    const serviceOrder = {name:'jose', birth_date:30, phone:30303030, email:'jose@gmail.com',cpf:123456, fantasName:''}

    const wrapper = mount(<CardClient serviceOrder={serviceOrder}/>)

    const findText = wrapper.find('Typography').at(6)
    expect(findText.text().split(":")[0]).toEqual("Nome Fantasia")    
  })

})
