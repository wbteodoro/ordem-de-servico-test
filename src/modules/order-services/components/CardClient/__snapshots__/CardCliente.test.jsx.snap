// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardClient Component should check snapshot version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Card)
    className={
      Object {
        "minWidth": 75,
      }
    }
  >
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        Nome: 
        jose
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        CPF: 
        123456
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        CNPJ: 
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        Telefone: 
        30303030
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        Data de Nascimento: 
        30
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        E-mail: 
        jose@gmail.com
      </WithStyles(Typography)>
      <WithStyles(Typography)
        className={
          Object {
            "fontSize": 15,
            "marginBottom": 12,
          }
        }
        color="textSecondary"
      >
        Nome Fantasia:
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </WithStyles(Card)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WithStyles(CardContent)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          Nome: 
          jose
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          CPF: 
          123456
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          CNPJ: 
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          Telefone: 
          30303030
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          Data de Nascimento: 
          30
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          E-mail: 
          jose@gmail.com
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className={
            Object {
              "fontSize": 15,
              "marginBottom": 12,
            }
          }
          color="textSecondary"
        >
          Nome Fantasia:
        </WithStyles(Typography)>
      </WithStyles(CardContent)>,
      "className": Object {
        "minWidth": 75,
      },
      "classes": Object {
        "root": "MuiCard-root-1",
      },
      "raised": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Nome: 
            jose
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            CPF: 
            123456
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            CNPJ: 
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Telefone: 
            30303030
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Data de Nascimento: 
            30
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            E-mail: 
            jose@gmail.com
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Nome Fantasia:
          </WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Nome: ",
              "jose",
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "Nome: ",
            "jose",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "CPF: ",
              123456,
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "CPF: ",
            123456,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "CNPJ: ",
              undefined,
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "CNPJ: ",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Telefone: ",
              30303030,
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "Telefone: ",
            30303030,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Data de Nascimento: ",
              30,
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "Data de Nascimento: ",
            30,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "E-mail: ",
              "jose@gmail.com",
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "E-mail: ",
            "jose@gmail.com",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Nome Fantasia:",
              undefined,
            ],
            "className": Object {
              "fontSize": 15,
              "marginBottom": 12,
            },
            "color": "textSecondary",
          },
          "ref": null,
          "rendered": Array [
            "Nome Fantasia:",
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(CardContent)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Nome: 
            jose
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            CPF: 
            123456
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            CNPJ: 
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Telefone: 
            30303030
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Data de Nascimento: 
            30
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            E-mail: 
            jose@gmail.com
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className={
              Object {
                "fontSize": 15,
                "marginBottom": 12,
              }
            }
            color="textSecondary"
          >
            Nome Fantasia:
          </WithStyles(Typography)>
        </WithStyles(CardContent)>,
        "className": Object {
          "minWidth": 75,
        },
        "classes": Object {
          "root": "MuiCard-root-1",
        },
        "raised": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              Nome: 
              jose
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              CPF: 
              123456
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              CNPJ: 
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              Telefone: 
              30303030
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              Data de Nascimento: 
              30
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              E-mail: 
              jose@gmail.com
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className={
                Object {
                  "fontSize": 15,
                  "marginBottom": 12,
                }
              }
              color="textSecondary"
            >
              Nome Fantasia:
            </WithStyles(Typography)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Nome: ",
                "jose",
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "Nome: ",
              "jose",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "CPF: ",
                123456,
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "CPF: ",
              123456,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "CNPJ: ",
                undefined,
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "CNPJ: ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Telefone: ",
                30303030,
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "Telefone: ",
              30303030,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Data de Nascimento: ",
                30,
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "Data de Nascimento: ",
              30,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "E-mail: ",
                "jose@gmail.com",
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "E-mail: ",
              "jose@gmail.com",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Nome Fantasia:",
                undefined,
              ],
              "className": Object {
                "fontSize": 15,
                "marginBottom": 12,
              },
              "color": "textSecondary",
            },
            "ref": null,
            "rendered": Array [
              "Nome Fantasia:",
              undefined,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
